cpp/src/arrow/ArrowConfig.cmake.in

```diff
@@ -96,11 +96,20 @@ include("${CMAKE_CURRENT_LIST_DIR}/ArrowTargets.cmake")
 
 if(TARGET Arrow::arrow_static AND NOT TARGET Arrow::arrow_bundled_dependencies)
   add_library(Arrow::arrow_bundled_dependencies STATIC IMPORTED)
-  get_target_property(arrow_static_location Arrow::arrow_static LOCATION)
-  get_filename_component(arrow_lib_dir "${arrow_static_location}" DIRECTORY)
+  get_target_property(arrow_static_debug_location Arrow::arrow_static
+                      IMPORTED_LOCATION_DEBUG)
+  get_filename_component(arrow_debug_lib_dir "${arrow_static_debug_location}" DIRECTORY)
   set_target_properties(Arrow::arrow_bundled_dependencies
                         PROPERTIES IMPORTED_LOCATION
-                                   "${arrow_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
+                                   "${arrow_debug_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  get_target_property(arrow_static_release_location Arrow::arrow_static
+                      IMPORTED_LOCATION_RELEASE)
+  get_filename_component(arrow_release_lib_dir "${arrow_static_release_location}"
+                         DIRECTORY)
+  set_target_properties(Arrow::arrow_bundled_dependencies
+                        PROPERTIES IMPORTED_LOCATION_RELEASE
+                                   "${arrow_release_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
```

Comment:
Can we use `IMPORTED_CONFIGURATIONS` instead of explicit `DEBUG` and `RELEASE`?

```diff
diff --git a/cpp/src/arrow/ArrowConfig.cmake.in b/cpp/src/arrow/ArrowConfig.cmake.in
index 5bd710893b..b2097a2f31 100644
--- a/cpp/src/arrow/ArrowConfig.cmake.in
+++ b/cpp/src/arrow/ArrowConfig.cmake.in
@@ -96,12 +96,16 @@ include("${CMAKE_CURRENT_LIST_DIR}/ArrowTargets.cmake")
 
 if(TARGET Arrow::arrow_static AND NOT TARGET Arrow::arrow_bundled_dependencies)
   add_library(Arrow::arrow_bundled_dependencies STATIC IMPORTED)
-  get_target_property(arrow_static_location Arrow::arrow_static LOCATION)
-  get_filename_component(arrow_lib_dir "${arrow_static_location}" DIRECTORY)
-  set_target_properties(Arrow::arrow_bundled_dependencies
-                        PROPERTIES IMPORTED_LOCATION
-                                   "${arrow_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
-  )
+  get_target_property(arrow_static_configurations Arrow::arrow_static IMPORTED_CONFIGURATIONS)
+  foreach(CONFIGURATION ${arrow_static_configurations})
+    string(TOUPPER "${CONFIGURATION}" CONFIGURATION)
+    get_target_property(arrow_static_location Arrow::arrow_static LOCATION_${CONFIGURATION})
+    get_filename_component(arrow_lib_dir "${arrow_static_location}" DIRECTORY)
+    set_target_properties(Arrow::arrow_bundled_dependencies
+                          PROPERTIES IMPORTED_LOCATION_${CONFIGURATION}
+                                     "${arrow_lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX}arrow_bundled_dependencies${CMAKE_STATIC_LIBRARY_SUFFIX}"
+    )
+  endforeach()
 
   # CMP0057: Support new if() IN_LIST operator.
   # https://cmake.org/cmake/help/latest/policy/CMP0057.html

```

Reply:
Sure, this PR is updated to use `IMPORTED_CONFIGURATIONS`

Reply:
Thanks.
Could you confirm this change with vcpkg?

Reply:
Thanks!